### PR TITLE
Remote layer setup

### PR DESCRIPTION
**Pull request by EslamHamouda**
Add Retrofit and Gson.
Create a Retrofit instance with the base URL.
Add Hilt to inject Retrofit instances.
Hide Base URL in gradle properties.

### CODE REVIEW

File: app/src/main/java/com/android/halanvendordiscovery/remote/RemoteClient.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import javax.inject.Named
```
Comment (KhaledSherifSayed) [inline app/src/main/java/com/android/halanvendordiscovery/remote/RemoteClient.kt]: reformate code and delete unused imports

File: app/src/main/java/com/android/halanvendordiscovery/remote/RemoteClient.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    fun provideGsonBuilder(): Gson {
+        return GsonBuilder()
+            .setLenient()
+            .create()
+    }
+
+    @Provides
+    @Singleton
+    fun provideLoggingInterceptor(): HttpLoggingInterceptor {
+        return HttpLoggingInterceptor()
```
Comment (KhaledSherifSayed) [inline app/src/main/java/com/android/halanvendordiscovery/remote/RemoteClient.kt]: change the code to only enable logging for debug builds

Comment (EslamHamouda) [reply app/src/main/java/com/android/halanvendordiscovery/remote/RemoteClient.kt]: Modified

**Review by KhaledSherifSayed — APPROVED** [review]
nice :rocket:

**Review by OmarMarei95 — APPROVED** [review]

**Pull request merged**